spring-shell2-core/src/main/java/org/springframework/shell2/commands/Help.java

```diff
@@ -148,8 +157,14 @@ else if (description.keys().size() > 1) {
 			if (description.defaultValue().isPresent()) {
 				result
 						.append("  [Optional, default = ", AttributedStyle.BOLD)
-						.append(description.defaultValue().get(), AttributedStyle.BOLD.italic())
-						.append("]", AttributedStyle.BOLD);
+						.append(description.defaultValue().get(), AttributedStyle.BOLD.italic());
+				description.defaultValueWhenFlag().ifPresent(
+					s -> result.append(", or ", AttributedStyle.BOLD)
+						.append(s, AttributedStyle.BOLD.italic())
+					.append(" if used as a flag", AttributedStyle.BOLD)
+				);
+
+				result.append("]", AttributedStyle.BOLD);
 			} else {
 				result.append("  [Mandatory]", AttributedStyle.BOLD);
```

Comment:
should this section consider the case when the key is mandatory but a defaultValueWhenFlag is present?

case in hand is the LegacyCommands::sum2 method, the v2 value, it currently says:
```
        --v2  [int]
                  [Mandatory]
```

it might be good if it specifies the value? for example:
[Mandatory, default = 42] ?